sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -98,6 +98,15 @@ trait CheckAnalysis {
 
             aggregateExprs.foreach(checkValidAggregateExpression)
 
+          case Sort(orders, _, _) =>
+            def checkValidSortOrder(order: SortOrder): Unit = order.dataType match {
```

Comment:
I think aggregate was only doing it because it was recursive. Not that big a of deal.